relay.go

```diff
@@ -50,37 +69,55 @@ func (r *Relayer) Hosts() RelayHosts {
 	return r.hosts
 }
 
-// Relay forwards a frame.
+// Relay is called for each frame that is read on the connection.
 func (r *Relayer) Relay(f *Frame) error {
 	if f.messageType() != messageTypeCallReq {
 		return r.handleNonCallReq(f)
 	}
 	return r.handleCallReq(f)
 }
 
-// Receive accepts a relayed frame.
-func (r *Relayer) Receive(f *Frame) {
-	r.RLock()
-	_, ok := r.items[f.Header.ID]
-	r.RUnlock()
-	if !ok {
-		r.logger.WithFields(
-			LogField{"ID", f.Header.ID},
-		).Warn("Received a frame without a RelayItem.")
+// Receive receives frames intended for this connection.
+func (r *Relayer) Receive(f *Frame, fType frameType) {
+	{
+		// NOTE, this block is safety checking. We should not enable this in
+		// production builds.
+
+		// If we receive a response frame, we expect to find that ID in our outbound.
+		// If we receive a request frame, we expect to find that ID in our inbound.
+		items := r.outbound
+		if fType == requestFrame {
+			items = r.inbound
+		}
```

Comment:
Yeah thought about this a few times. I didn't want a generic method since it actually depends on whether you're the receiver or it originated on your connection. Put it into a method, but made it specific `receiverItems`